packages/react-noop-renderer/src/createReactNoop.js

```diff
@@ -913,7 +913,7 @@ function createReactNoop(reconciler: Function, useMutation: boolean) {
       }
     },
 
-    flushSync(fn: () => mixed) {
+    flushSync<T>(fn: ?() => T): ?T {
```

Comment:
I don't know how to do function overloading in Flow, but it needs to be that if a function is passed, the return type is the same as the return type of the function. As-is, the caller would need to check if the result is undefined.

Reply:
Yeah good catch. I pushed a way to do it that works.

## Before
<img width="986" alt="Screen Shot 2021-07-15 at 3 46 19 PM" src="https://user-images.githubusercontent.com/2440089/125850380-7cdeb053-cf09-4d9d-9565-667eb79208b8.png">

## After
<img width="986" alt="Screen Shot 2021-07-15 at 4 03 51 PM" src="https://user-images.githubusercontent.com/2440089/125850394-a0117e4d-f217-4ce5-bf96-b0e99cb9d76a.png">

Reply:
Note, I tried using an intersection type but that results in some really cryptic errors like "R does not match return type R" so I opted for this version instead.